### PR TITLE
make AddLabel workflow work as expected

### PR DESCRIPTION
**Pull request by charlesli640**
# Description

Make Add Label workflow work - adding Label "pull ready" when PR got approved.

 1.Make AddLabel workflow in chain of "MaxText Package Tests".
 2.Make AddLabel workflow to fail by `core.setFailed + return;` when condition is not met
 3.Need >=2 approvals (with different approvers) to set the "pull ready" label

# Tests

CI/CD

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: .github/workflows/AddLabel.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ name: Add Label
 
 on:
   workflow_run:
-    workflows: [Tests, CodeQL]
```
Comment (charlesli640) [inline .github/workflows/AddLabel.yml:19]: Only need chain CI Test workflow "MaxText Package Tests"

File: .github/workflows/AddLabel.yml [hunk 2]
```diff
@@ -68,15 +69,15 @@ jobs:
...
             }
             let is_approved = false
             if (decision_result.repository.pullRequest.reviewDecision === "APPROVED") {
               is_approved = true
             }
             if (!is_approved) {
               console.log("Not adding pull ready because the PR is not approved yet by sufficient code owners.")
-              process.exit()
+              process.exit(1)
             }
```
Comment (MichaelHudgins) [inline .github/workflows/AddLabel.yml]: Would all of this logic be better handled by making this a condition of the job executing at all.

```
if: github.event.review.state == 'approved'
```

See https://docs.github.com/en/actions/reference/workflows-and-actions/events-that-trigger-workflows#running-a-workflow-when-a-pull-request-is-approved

File: .github/workflows/AddLabel.yml [hunk 3]
```diff
@@ -85,11 +86,11 @@ jobs:
...
               per_page: 100,
             })
-            // Check that the number of commits in the PR is 1.
-            if (commits.data.length !== 1) {
-              console.log("Not adding pull ready because the PR has more than one commit. Please squash your commits.")
-              process.exit(1)
-            }
+            // Check that the number of commits in the PR is 1. This is not enforced in practice.
+            // if (commits.data.length !== 1) {
+            //  console.log("Not adding pull ready because the PR has more than one commit. Please squash your commits.")
```
Comment (MichaelHudgins) [inline .github/workflows/AddLabel.yml]: I recommend leaving instructions or a link on how to do this for users.  

https://docs.jax.dev/en/latest/contributing.html#single-change-commits-and-pull-requests is how jax lists it

File: .github/workflows/AddLabel.yml [hunk 4]
```diff
@@ -85,11 +86,11 @@ jobs:
               pull_number,
               per_page: 100,
             })
-            // Check that the number of commits in the PR is 1.
-            if (commits.data.length !== 1) {
-              console.log("Not adding pull ready because the PR has more than one commit. Please squash your commits.")
-              process.exit(1)
-            }
+            // Check that the number of commits in the PR is 1. This is not enforced in practice.
```
Comment (MichaelHudgins) [inline .github/workflows/AddLabel.yml]: I also recommend adding a bypass label for checks like this.  Allow the reviewer to basically flag this PR is ok to have multi commits.

File: .github/workflows/AddLabel.yml [hunk 5]
```diff
@@ -98,13 +99,13 @@ jobs:
...
             }
             for (const checkRun of checkRuns.data.check_runs) {
               if (checkRun.name.endsWith(context.job)) continue
               if (checkRun.conclusion !== "success") {
                 console.log("Not adding pull ready because " + checkRun.name + " has not passed yet: " + checkRun.html_url)
-                process.exit()
+                process.exit(1)
               }
             }
             console.log("Adding pull ready label because the PR is approved AND all the check runs have passed")
```
Comment (MichaelHudgins) [inline .github/workflows/AddLabel.yml:121]: This job seems like it should be doing some extra verification on who the approver is before adding the label.

File: .github/workflows/AddLabel.yml (hunk 2 above)
Comment (charlesli640) [reply .github/workflows/AddLabel.yml]: Agree. That logic is more concise.

File: .github/workflows/AddLabel.yml (hunk 5 above)
Comment (charlesli640) [reply .github/workflows/AddLabel.yml:121]: Actually not all people can approve PRs. And the approvers are kind of dynamic as people may join/leave the project. I am afraid it is not necessary to check who approved the PR. 
Even we check who approved PR, what is the next step? What is the criteria to give it a green light or red light?

File: .github/workflows/AddLabel.yml (hunk 3 above)
Comment (charlesli640) [reply .github/workflows/AddLabel.yml]: If forcing single-commit is the rule, let's follow the rule.

File: .github/workflows/AddLabel.yml (hunk 4 above)
Comment (charlesli640) [reply .github/workflows/AddLabel.yml]: Actually don't want to make it too complicated for this simple workflow.
If multi-commit is acceptable in special case, approver can manually add "pull ready" label to specially treat that PR.

File: .github/workflows/AddLabel.yml (hunk 2 above)
Comment (charlesli640) [reply .github/workflows/AddLabel.yml]: The error github.event.review not found (or being empty) will occur when workflow is triggered by an event that does not include a review object in its payload. Cannot use github.event.review.state directly without checking.

File: .github/workflows/AddLabel.yml [hunk 6]
```diff
@@ -16,9 +16,8 @@ name: Add Label
 
 on:
   workflow_run:
-    workflows: [Tests, CodeQL]
-    types:
-      - completed
+    workflows: ["MaxText Package Tests"]
```
Comment (charlesli640) [inline .github/workflows/AddLabel.yml:19]: Comments test

**Review by shralex — APPROVED** [review]

**Review by shralex — APPROVED** [review]

File: .github/workflows/AddLabel.yml (hunk 6 above)
Comment (charlesli640) [reply .github/workflows/AddLabel.yml:19]: comments trigger test.

**Review by khatwanimohit — APPROVED** [review]
